ionoscloud/resource_server.go

```diff
@@ -160,15 +170,48 @@ func resourceServer() *schema.Resource {
 							Optional:   true,
 							Deprecated: "Please use ssh_key_path under server level",
 							Computed:   true,
+							ForceNew:   true,
 							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
I've tested creating a server with a volume which has ssh_key_path and it's fine, so there is no need of DiffSuppressFunc on volume level. We want to use DiffSuppressFunc only when we have a field that trying to change in plan before we want to make terraform apply, but we know this change will not affect the infrastructure described in the plan.

Reply:
@gabisavu-ionos not sure what you mean by `"I've tested creating a server with a volume which has ssh_key_path and it's fine"`, I didn't mention something about functionality (I assumed already that you tested and it's working). 

> so there is no need of DiffSuppressFunc on volume level

I was speaking about adding this `DiffSuppressFunc` for **server's** `ssh_keys` and `ssh_key_path`, not on volume level.

> only when we have a field that trying to change in plan before we want to make terraform apply, but we know this change will not affect the infrastructure described in the plan.

I didn't understand this part, can you please rephrase? 

And, to be more clear, when I wrote this comment I did it because I noticed that we have `DiffSuppressFunc` for `ssh_keys` and `ssh_key_path`, so I assumed that, for these fields, we need to have this function (regardless of where the fields are located: on volume level or on server level). Right now, if I provide `ssh_keys` in the tf plan on the **volume** resource within a server resource,  then `DiffSuppressFunc` is called and an extra check is done inside this function, but if I provide `ssh_keys` only on the **server** resource, then this function is **not** called and the check is **not** done and I'm not sure if we want this. It made more sense to me to use the `DiffSuppressFunc` in both cases, since we are talking about the same fields: `ssh_keys`/`ssh_key_path`.

Reply:
@gabisavu-ionos I mark this as `unsolved` again, let's have a quick call when you can to discuss the details.

Reply:
Marking this as solved again, I discussed with  @gabisavu-ionos and we decided to leave this as it is since the `DiffSuppressFunc` will be removed in the future. Besides that, right now, the function is not used properly, since it is more of a workaround to see if the `ssh_keys` order is the same for the keys set on `volume` and `server` levels.